File: foxy/parent/src/command/vanilla/entertainment/memes/StonksExecutor.ts
import { ExecutorParams } from "../../../structures/CommandExecutor";
import { bot } from "../../../../FoxyLauncher";

export default class StonksExecutor {
    async execute({ context, endCommand }: ExecutorParams) {
        context.sendDefer();
        const content = context.getOption<string>("text", false);

        const stonksImage = await bot.rest.foxy.getArtistryImage("/memes/stonks", {
            text: content
        });

        const file = new File([stonksImage], "stonks.png", { type: "image/png" });

        context.reply({
            file: {
                name: "stonks.png",
                blob: file
            }
        });

        return endCommand();
    }
}

File: foxy/src/main/kotlin/net/cakeyfox/foxy/interactions/vanilla/magic/StonksExecutor.kt
package net.cakeyfox.foxy.interactions.vanilla.magic

import io.ktor.client.call.*
import io.ktor.utils.io.*
import kotlinx.coroutines.withContext
import kotlinx.io.readByteArray
import kotlinx.serialization.json.buildJsonObject
import kotlinx.serialization.json.put
import net.cakeyfox.foxy.interactions.commands.CommandContext
import net.cakeyfox.foxy.interactions.commands.UnleashedCommandExecutor
import net.dv8tion.jda.api.utils.FileUpload

class StonksExecutor : UnleashedCommandExecutor() {
    override suspend fun execute(context: CommandContext) {
        context.defer()
        val text = context.getOption("text", 0, String::class.java, true)

        val stonksImage = withContext(context.foxy.coroutineDispatcher) {
            context.foxy.artistryClient.generateImage("memes/stonks", buildJsonObject {
                put("text", text)
            })
        }

        if (stonksImage.status.value !in 200..299) {
            throw IllegalArgumentException("Error while generating image! Received ${stonksImage.status}")
        }

        val response = stonksImage.body<ByteReadChannel>().readRemaining().readByteArray()
        val imageAsFile = FileUpload.fromData(response, "stonks.png")

        context.reply {
            files.plusAssign(imageAsFile)
        }
    }
}